intake/catalog/remote.py

```diff
@@ -26,30 +27,38 @@ def describe(self):
     def get(self, **user_parameters):
         entry = self.kwargs
 
-        return RemoteDataSource(self.url, entry['name'], container=entry['container'], user_parameters=user_parameters, description=entry['description'])
+        return RemoteDataSource(
+            self.url, entry['name'], container=entry['container'],
+            user_parameters=user_parameters, description=entry['description'])
 
 
 class RemoteDataSource(DataSource):
-    def __init__(self, url, entry_name, container, user_parameters, description):
-        self._init_args = dict(url=url, entry_name=entry_name, container=container, user_parameters=user_parameters, description=description)
+    def __init__(self, url, entry_name, container, user_parameters,
+                 description):
+        self._init_args = dict(
+            url=url, entry_name=entry_name, container=container,
+            user_parameters=user_parameters, description=description)
 
         self._url = url
         self._entry_name = entry_name
-        self._user_parameters = user_parameters
+        self._user_parameters = expand_templates(
+            user_parameters, template_context=TemplateContext(None))
 
         self._real_source = None
         self.direct = None
 
-        super(RemoteDataSource, self).__init__(container=container, description=description)
+        super(RemoteDataSource, self).__init__(container=container,
+                                               description=description)
 
     def _open_source(self):
         if self._real_source is None:
             payload = dict(action='open',
                            name=self._entry_name,
                            parameters=self._user_parameters,
                            available_plugins=list(plugin_registry.keys()))
-            req = requests.post(self._url, data=msgpack.packb(payload, use_bin_type=True))
-            if req.status_code == 200:
+            req = requests.post(self._url, data=msgpack.packb(
+                payload, use_bin_type=True))
+            if req.ok:
```

Comment:
For anyone interested, this changed from code==200, since in theory the server should be able to do redirects, and perhaps at some later date administrative actions too.